src_python/yaafelib/audiofeature.py

```diff
@@ -25,14 +25,32 @@
 import yaafelib.core as yaafecore
 
 
+# Taken from python package `six`
+# https://bitbucket.org/gutworth/six/src/2c12cd64ff0c7797bb30b0d466e902f7ecd6e562/six.py?fileviewer=file-view-default#six.py-826
+def add_metaclass(metaclass):
+    """Class decorator for creating a class with a metaclass."""
+    def wrapper(cls):
+        orig_vars = cls.__dict__.copy()
+        slots = orig_vars.get('__slots__')
+        if slots is not None:
+            if isinstance(slots, str):
+                slots = [slots]
+            for slots_var in slots:
+                orig_vars.pop(slots_var)
+        orig_vars.pop('__dict__', None)
+        orig_vars.pop('__weakref__', None)
+        return metaclass(cls.__name__, cls.__bases__, orig_vars)
+    return wrapper
+
+
 def check_dataflow_params(f):
     def _safe_get_dataflow(cls, params, samplerate):
         # filter parameters and set missing parameters with default values
-        for k, v in params.iteritems():
+        for k, v in params.items():
```

Comment:
I was wondering whether you wanted to depend on `six` or not, hence decided to use this method :)

I'll use `six` then and remove the copied padd_metaclass](https://github.com/Yaafe/Yaafe/pull/21/files/8716b5465fb9392f3da78ab062d9ea42dba5d650#diff-92c35c322736fe6a2a98bf3ec3109fdbR30)

The major concern I had was how do I make sure that when a user is installing `yaafe`, six gets installed too ? (As six is a required dependency now)
I normally use pip for that for the dependency management ... Any ideas ?

To keep it simple, and not depend on six I could also do `(key, params[key] for key in params)` which would act as a generator in both
```
```